wallet/wallet.h

```diff
@@ -365,10 +365,6 @@ enum invoice_status {
 };
 
 struct invoice {
-	/* List off ld->wallet->invoices */
-	struct list_node list;
```

Comment:
OK, this is subtle: memleak can't find things in lists which aren't at the head.  We could change that but it's a lot of work, so this is why you can't find the memleak!

Reply:
..... oh, I see....